docs/src/guides/scalar_index.md

```diff
@@ -0,0 +1,109 @@
+# Building Scalar Index
+
+Similar to many SQL databases, LanceDB supports several types of Scalar indices to accelerate search
+over scalar columns.
+
+- `BTREE`: The most common type is BTREE. This index is inspired by the btree data structure
+  although only the first few layers of the btree are cached in memory.
+  It will perform well on columns with a large number of unique values and few rows per value.
+- `BITMAP`: this index stores a bitmap for each unique value in the column.
+  This index is useful for columns with a small number of unique values and many rows per value.
+  For example, a column presents `categories`. In Apache Arrow terminology, a column can be presented
+  as `DictionaryType`.
+- `LABEL_LIST`: a special index that is used to index list columns whose values have small cardinality.
+  For example, a column that contains lists of tags (e.g. `["tag1", "tag2", "tag3"]`) can be indexed with a `LABEL_LIST` index.
+
+| Data Type                                     | Filter                           | Index Type   |
+| --------------------------------------------- | -------------------------------- | ------------ |
+| Numeric, String, Boolean                      | `<`, `=`, `>`                    | `BTREE`      |
+| Low cardinality of numbers or strings         | `<`, `=`, `>`, `in`, `between`   | `BITMAP`     |
+| List of low cardinality of numbers or strings | `array_has_any`, `array_has_all` | `LABEL_LIST` |
+
+=== "Python"
+
+    ```python
+    import lancedb
+    data = [
+      {"my_col": 1, "category": "A", "vector": [1, 2]},
+      {"my_col": 2, "category": "B", "vector": [3, 4]},
+      {"my_col": 3, "category": "C", "vector": [5, 6]}
+    ]
+
+    db = lancedb.connect("./db")
+    table = db.create_table("my_table", data)
+    table.create_scalar_index("my_col")  # BTree by default
+    table.create_scalar_index("category", index_type="BITMAP")
+    ```
+
+=== "Typescript"
+
+    Only `BTree` index is supported today. `BITMAP` and `LABEL_LIST` will be added soon.
+
+    Follow [Github issue #1511](https://github.com/lancedb/lancedb/issues/1511) for updates.
+
+    ```js
+    const db = await lancedb.connect("data/sample-lancedb");
+    const tbl = await db.openTable("my_vectors");
+
+    await tbl.create_index("my_float_column");
+    ```
+
+For example, the following scan will be faster if the column `my_col` has a scalar index:
+
+=== "Python"
+
+    ```python
+    import lancedb
+
+    table = db.open_table("my_table")
+    my_df = table.search().where("my_col = 2").to_pandas()
+    ```
+
+=== "Typescript"
+
+    === "@lancedb/lancedb"
+
+        ```js
+        const db = await lancedb.connect("data/lance");
+        const tbl = await db.openTable("images");
+
+        await tbl
+          .query()
+          .where("my_col != 7")
+          .limit(10)
+          .toArray();
+        ```
+
+Scalar indices can also speed up scans containing a vector search or full text search, and a prefilter:
```

Comment:
Why mention `full text search` here?  I'm not entirely sure what this would imply.  I don't think we ever connected the "inverted scalar index" (which isn't mentioned here) with the `LIKE` operator.

Reply:
I meant to use pre-filter as part of the FTS.

Reply:
I can just remove it

Reply:
Ah, no, that makes sense.

Reply:
Go ahead and keep it